packages/expo-cli/src/accounts.ts

```diff
@@ -95,7 +110,165 @@ export async function login(options: CommandOptions): Promise<User> {
   }
 }
 
-async function _usernamePasswordAuth(username?: string, password?: string): Promise<User> {
+/**
+ * Prompt for an OTP with the option to cancel the question by answering empty (pressing return key).
+ */
+async function _promptForOTPAsync(cancelBehavior: 'cancel' | 'menu'): Promise<string | null> {
+  const enterMessage =
+    cancelBehavior === 'cancel' ? 'press Enter to cancel' : 'press Enter for other options';
+  const otpQuestion: Question = {
+    type: 'input',
+    name: 'otp',
+    message: `One-time Password or Backup Code (${enterMessage}):`,
+  };
+
+  const { otp } = await prompt(otpQuestion);
+  if (!otp) {
+    return null;
+  }
+
+  return otp;
+}
+
+/**
+ * Prompt for user to choose a backup OTP method. If selected method is SMS, a request
+ * for a new OTP will be sent to that method. Then, prompt for the OTP, and retry the user login.
+ */
+async function _promptForBackupOTPAsync(
+  username: string,
+  password: string,
+  secondFactorDevices: SecondFactorDevice[]
+): Promise<string | null> {
+  const nonPrimarySecondFactorDevices = secondFactorDevices.filter(device => !device.is_primary);
+
+  if (nonPrimarySecondFactorDevices.length === 0) {
+    throw new Error('No other second-factor devices set up');
+  }
+
+  const hasNonAuthenticatorSecondFactorDevice = nonPrimarySecondFactorDevices.find(
+    device => device.method !== UserSecondFactorDeviceMethod.AUTHENTICATOR
+  );
```

Comment:
hmm, this logic is incorrect now. This should be `hasAuthenticatorSecondFactorDevice` and the condition should be inverted.